protos/sigstore_verification.proto

```diff
@@ -0,0 +1,82 @@
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+package dev.sigstore.verification.v1;
+
+import "sigstore_common.proto";
+import "sigstore_trustroot.proto";
+import "sigstore_bundle.proto";
+
+option go_package = "github.com/sigstore/proto/verification/v1";
+option java_package = "dev.sigstore.proto.verification.v1";
+option java_multiple_files = true;
+option java_outer_classname = "VerificationProto";
+
+// The identity of a signer.
+message Identity {
+        // The X.509v3 issuer extension (OID 1.3.6.1.4.1.57264.1.1)
+        string issuer = 1;
+        // A regular expression describing the expected value for the SAN.
+        string subject_alternative_name = 2;
+}
+
+message TransparencyLogs {
+        // The set of transparency logs to use during bundle verification.
+        repeated dev.sigstore.trustroot.v1.TransparencyLogInstance instances = 1;
+        // The minimim number of logs the entry must appear on.
+        // Setting this value to zero disables verification.
+        int32 threshold = 2;
+        // Perform an online proof of inclusion.
+        bool require_inclusion_proof = 3;
```

Comment:
Yea, I think this requires some more thought. As of now, an inclusion proof is really no different than a promise, since it doesn't verify consistency. And even if you do verify consistency from a witness, now you need a threshold of witness signatures, but this isn't supported by Rekor currently (it's tracked as an issue for a 2.0 rekor)

I'd probably say to remove this entirely - I can create a tracking issue for what this could look like.

Reply:
The online verification option is moved, with a comment on this. Resolving.